src/main/java/core/basesyntax/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.FruitTransaction;
+import core.basesyntax.strategy.operation.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(
+            Map<FruitTransaction.Operation, OperationHandler> operationHandlers
+    ) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public OperationHandler getOperationHandler(FruitTransaction.Operation operation) {
+        return operationHandlers.get(operation);
```

Comment:
better add a null check for the handler. and throw an exception if the handler doesn’t  exist